### PR TITLE
[84] On writes call rcDataObjOpen() with O_CREAT flag

### PR DESCRIPTION
**Pull request by JustinKyleJames**
instead of calling rcDataObjOpen followed by rcDataObjCreate if the open fails

### CODE REVIEW

File: DSI/globus_gridftp_server_iRODS.cpp [hunk 1]
```diff
@@ -2023,62 +2022,10 @@ globus_l_gfs_iRODS_recv(
...
-
-                fd_info = nlohmann::json::parse(j_out);
-                std::string replica_token = fd_info.at("replica_token").get<std::string>();
-                iRODS_handle->replica_token = strdup(replica_token.c_str());
-            }
-        }
+            result = globus_l_gfs_iRODS_make_error("rcDataObjOpen failed", iRODS_handle->fd);
+            globus_gridftp_server_finished_transfer(op, result);
+            return;
+	}
```
Comment (korydraughn) [inline DSI/globus_gridftp_server_iRODS.cpp:2028]: Indentation

**Review by korydraughn — COMMENTED** [review]
Very nice.

I second what @alanking said.

File: DSI/globus_gridftp_server_iRODS.cpp (hunk 1 above)
Comment (JustinKyleJames) [reply DSI/globus_gridftp_server_iRODS.cpp]: > Indentation

Done

**Comment by korydraughn** [comment]
Ok. Since there's no support for automated tests right now, the only question left is ... have you verified the changes work as intended?

**Comment by alanking** [comment]
Ah, I see. No automated tests at this point.

The changes seem good to me if it works.

**Comment by JustinKyleJames** [comment]
> Ok. Since there's no support for automated tests right now, the only question left is ... have you verified the changes work as intended?

Yes, I have done manual testing on the change.

**Comment by korydraughn** [comment]
Squash it.

**Review by alanking — APPROVED** [review]
Yep, # it.

**Review by korydraughn — APPROVED** [review]
Cool cool. Pound it.

**Comment by JustinKyleJames** [comment]
> Cool cool. Pound it.

Done

**Pull request merged**